pkg/build/builder/cmd/builder.go

```diff
@@ -107,19 +111,33 @@ func setupSourceSecret(sourceSecretName string, scmAuths []scmauth.SCMAuth) erro
 	}
 
 	if len(scmAuthsPresent) == 0 {
-		return fmt.Errorf("no auth handler was found for the provided secret %q",
+		return nil, fmt.Errorf("no auth handler was found for the provided secret %q",
 			sourceSecretName)
 	}
 
+	var urlArray []*url.URL
 	for name, auth := range scmAuthsPresent {
 		glog.V(3).Infof("Setting up SCMAuth %q", name)
-		if err := auth.Setup(sourceSecretDir); err != nil {
+		u, err := auth.Setup(sourceSecretDir)
+		if err != nil {
 			// If an error occurs during setup, fail the build
-			return fmt.Errorf("cannot set up source authentication method %q: %v", name, err)
+			return nil, fmt.Errorf("cannot set up source authentication method %q: %v", name, err)
+		}
+		if u != nil {
+			urlArray = append(urlArray, u)
 		}
 	}
 
-	return nil
+	if len(urlArray) == 0 {
+		return nil, nil
+	}
+
+	for i := range urlArray {
+		if urlArray[i].String() != urlArray[0].String() {
+			return nil, fmt.Errorf("secret handlers contain different source URLs defined for the secret %q", sourceSecretName)
```

Comment:
none of our secret methods return URL overrides containing passwords, which is why I'm ok logging these